### PR TITLE
wr executor

### PR DESCRIPTION
**Pull request by sb10**
Work in progress for an executor that uses wr as a scheduler. See https://github.com/nextflow-io/nextflow/issues/1088

### CODE REVIEW

File: modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+		log.debug "[wr] cacertpath=$path"
+        return path
+    }
+
+    /**
+     * @return {@code false} whenever the containerization is managed by the executor itself
+     */
+    boolean isContainerNative() {
+        return false
+    }
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy]: Does `wr` manage the container execution on its own? In that case, it should be `true`.

File: modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy [hunk 2]
```diff
@@ -0,0 +1,664 @@
...
+@Slf4j
+@CompileStatic
+class WrTaskHandler extends TaskHandler implements BatchHandler<String,Map> {
+
+    // static final private String WORK_DIR = '/work'
+
+    final List<String> COMPLETE_STATUSES = ['complete', 'buried', 'deleted']
+
+    final List<String> STARTED_STATUSES = ['delayed', 'reserved', 'running'] + COMPLETE_STATUSES
+
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy]: Makes sense to declare these fields also `public static`

File: modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+@Slf4j
+@CompileStatic
+class WrExecutor extends Executor {
+
+	static private String token
+	static private String defaultManagerDir
+    static private String endpoint
+	static private String cacertPath
+	static private WrRestApi client
+
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy]: It looks there's a problem with tab/space indentation. Please use 4 spaces in place of tabs. More details [here](https://github.com/nextflow-io/nextflow/blob/master/CONTRIBUTING.md).

File: modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy (hunk 1 above)
Comment (sb10) [reply modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy]: As in doing something if the user wishes to run a docker container? No, wr doesn't yet have native container support. I'm hoping BashWrapperBuilder will put something appropriate in the wrapper script if this boolean returns false.

Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy]: Then it's fine and yes, BashWrapperBuilder should manage the container as usual.

**Comment by sb10** [comment]
After I `make compile` I end up with:

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   docs/conf.py
	modified:   modules/nextflow/src/main/groovy/nextflow/Const.groovy
	modified:   nextflow
```

I did not modify these files myself,
Is this expected? Is there a way to avoid these modifications being reported by git? And to avoid conflicts with them etc?

File: modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy (hunk 2 above)
Comment (sb10) [reply modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy]: I made that change, thanks.

File: modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy (hunk 3 above)
Comment (sb10) [reply modules/nextflow/src/main/groovy/nextflow/executor/WrExecutor.groovy]: Should now be fixed

**Comment by pditommaso** [comment]
The `compile` automatically changes the timestamp and build number, but those changes should not be commited.

**Comment by sb10** [comment]
Superseded by #1148

**Pull request closed without merging**